hyperspy/io.py

```diff
@@ -299,11 +303,18 @@ def load_single_file(filename, **kwds):
 
     """
     extension = os.path.splitext(filename)[1][1:]
-
     i = 0
-    while extension.lower() not in io_plugins[i].file_extensions and \
-            i < len(io_plugins) - 1:
-        i += 1
+    
+    if preferred_format:
+        for index,plugin in enumerate(io_plugins):
+           if extension.lower() in plugin.file_extensions and \
+                preferred_format.lower() in plugin.format_name.lower():
+               i=index
```

Comment:
ah - I suppose the issue with hdf5_format instead of preferred_format is that actually the user could specify any format and try to use that reader to load some file. Given that,  I think it might be best to go with just "format"?

Reply:
format is python function so better to use alternative name ?   io_format ?

Reply:
Good point.... file_format?

Reply:
changed to file_format